service/src/main/java/com/epam/rft/atsy/service/impl/StateFlowServiceImpl.java

```diff
@@ -47,4 +49,16 @@
 
     return converterService.convert(stateFlowEntities, StateFlowDTO.class);
   }
+  @Transactional(readOnly = true)
+  @Override
+  public boolean isAvailableFromLastState(List<StateHistoryDTO> representationList, String state) {
+    Collection<StateFlowDTO> available = getStateFlowDTOByFromStateDTO(representationList.get(0).getStateDTO());
```

Comment:
`getStateFlowDTOByFromStateDTO` includes a bunch of conversions, you can introduce a new repository method which can query the database if there is an existing StateFlowEntity which has references to the last state and the new state.